### PR TITLE
Us 31 admin merchants top merchants best day

### LINKED ISSUE #32: US-31-admin-merchants-top-merchants-best-day
31. Admin Merchants: Top Merchant's Best Day

As an admin,
When I visit the admin merchants index
Then next to each of the 5 merchants by revenue I see the date with the most revenue for each merchant.
And I see a label “Top selling date for <merchant name> was <date with most sales>"

Note: use the invoice date. If there are multiple days with equal number of sales, return the most recent day.

### PR DESCRIPTION
**Pull request by kohljd**
## Type of change
- [x] New feature
- [ ] Bug Fix
- [ ] Test
- [ ] Refactor/Style

## Description
Relevant User Story(s): 31

Display each top 5 merchant's largest total revenue date.
100% model and feature coverage per simplecov. 

## Implements/Fixes Issue(s):
closes #32 

## Neccesary checkmarks:
- [x] All Tests are Passing
- [x] The code will run locally

## Any Outstanding Issues/Problems?
- [x] My code is wonderful, no issues here!
- [ ] Yeah, see additional words below
*Description of problem(s)*

### CODE REVIEW

File: app/models/merchant.rb [hunk 1]
```diff
@@ -39,4 +39,15 @@ def self.top_five_merchants
       ORDER BY sum(invoice_items.unit_price * invoice_items.quantity) DESC
       LIMIT 5")
   end
-end
+
+  def top_sales_day  
+    self.invoices.joins(:transactions, :invoice_items)
+      .where("transactions.result = 0")
+      .select("invoices.created_at, sum(invoice_items.quantity * invoice_items.unit_price) AS invoice_revenue")
```
Comment (yainporter) [inline app/models/merchant.rb:46]: Do we need to have successful transactions? If so, this method won't work, you are duplicating data by joining to the invoice_items table twice

File: spec/models/merchant_spec.rb [hunk 2]
```diff
@@ -142,5 +142,30 @@
...
+        create(:transaction, invoice: invoice_8)
+        invoice_item_1 = create(:invoice_item, item: item_5, invoice: invoice_7, quantity: 1, unit_price: 1000)
+        invoice_item_2 = create(:invoice_item, item: item_5, invoice: invoice_8, quantity: 2, unit_price: 1500)
+
+        merchant_3 = create(:merchant)
+        item_6 = create(:item, merchant: merchant_3)
+        invoice_9 = create(:invoice, created_at: "2022-03-08")
+        invoice_10 = create(:invoice, created_at: "2023-05-19")
+        create(:transaction, invoice: invoice_9)
+        create(:transaction, invoice: invoice_10)
```
Comment (yainporter) [inline spec/models/merchant_spec.rb:162]: Duplicate the transactions of one that is supposed to be less, and see what I mean about duplicating data when joining the transactions table

**Review by yainporter — CHANGES_REQUESTED** [review]
Do we really need to count successful transactions?

**Comment by yainporter** [comment]
Could we add in more testing for the model method to check that even if something has multiple transactions, it will still be the one on top? The transactions here only appear once.

File: app/models/merchant.rb (hunk 1 above)
Comment (JRIV-10) [reply app/models/merchant.rb:46]: As discussed in our stand up this issue has been resloved!

**Review by JRIV-10 — APPROVED** [review]

**Pull request merged**